sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -946,6 +947,45 @@ object TransposeWindow extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * Replaces duplicate window expressions with an alias in a Project above the Window node.
```

Comment:
Looks like this is the case that the logic of sub common exprs should handle? cc: @viirya @cloud-fan

Reply:
Isn't the subexpression elimination wholestage codegen specific?

Reply:
The interpreted mode has the optimization, too, e.g., `SubExprEvaluationRuntime`.

Reply:
I see, thanks for the reference.
Indeed, the common subexpression elimination seems to eliminate these duplicates (if I did follow the code correctly).
I would argue, that there still is benefit in removing these duplicates in the optimizer. If nothing else, then it at least would clean up the plan.
Also the subexpression cache has a limited size (although configurable) and this would reduce the runtime overhead of cache lookups.